docker/entrypoint.sh

```diff
@@ -162,79 +228,77 @@ cat /tmp/certbot-hosts.txt | xargs -n 1 -I {} oc process -f /tmp/certbot-route.y
 # Sleep for 5sec. There was an issue noticed where the pod wasn't able to get a route and was giving 404 error. Not totally certain if this helps.
 sleep 5s
 
-rm -f $CERTBOT_WORK_DIR/deployed
-CERTBOT_ARGS='--no-random-sleep --no-eff-email'
-if [ "${CERTBOT_DRY_RUN}" == "true" ]; then
-  CERTBOT_ARGS="${CERTBOT_ARGS} --dry-run"
-fi
+rm -f ${CERTBOT_WORK_DIR}/deployed
 
-if [ "${CERTBOT_DEBUG}" == "true" ]; then
-  CERTBOT_ARGS="${CERTBOT_ARGS} --debug"
-fi
-
-if [ "${CERTBOT_SUBSET}" == "true" ]; then
-  CERTBOT_ARGS="${CERTBOT_ARGS} --allow-subset-of-names"
-fi
-
-if [ ! -z "$CERTBOT_SERVER" ]; then
-  CERTBOT_ARGS="${CERTBOT_ARGS} --server ${CERTBOT_SERVER}"
-fi
-
-set -x
-# if there is no certificate issue, request a new one
-if [ ! -f "${CERTBOT_CONFIG_DIR}/live/openshift-route-certs/cert.pem" ]; then
-  certbot --config /tmp/certbot.ini certonly $CERTBOT_ARGS --non-interactive --keep-until-expiring --cert-name 'openshift-route-certs' --expand --standalone -d "$(</tmp/certbot-hosts.csv)"
+# Get certificate(s), either combined or individual
+if [ "${CERTBOT_CERT_PER_HOST}" == "true" ]; then
+  echo "Manage individual certificates for each unique host."  
+  for certbot_host in $(</tmp/certbot-hosts.txt); do
+    getCertificate "${certbot_host}" "${certbot_host}"
+  done
 else
-  # if a certificate already exists, request to renew it
-  certbot --config /tmp/certbot.ini renew $CERTBOT_ARGS --no-random-sleep-on-renew --cert-name 'openshift-route-certs'
-fi
-set +x
+  echo "Managing a single certificate covering all managed hosts."
+  getCertificate "openshift-route-certs" "$(</tmp/certbot-hosts.csv)"
```

Comment:
Wouldn't be a bad idea to make `"openshift-route-certs"` be a defined defaultName variable of some kind in the shell script.

Reply:
Done